src/request.ts

```diff
@@ -19,7 +19,7 @@ export class FilterRegistry extends Registry<RequestFilter> {
 		}
 		else if (test instanceof RegExp) {
 			entryTest = (response, url, options) => {
-				return test.test(url);
+				return test ? (<RegExp> test).test(url) : null;
```

Comment:
You could work around this guard being reset (and the subsequent need to cast) by doing something like this at the top of the function scope:

```typescript
const inTest = test;
```

The TypeScript team are likely to add the ability to declare arguments as `const` though so that type narrowing doesn't get reset in function blocks like this.